geoutils/raster/raster.py

```diff
@@ -3363,96 +3443,186 @@ def split_bands(self: RasterType, copy: bool = False, bands: list[int] | int | N
+        that will be used as data column of the point cloud).
+        If 'subsample' is either 1, or is equal to the pixel count, all valid points are returned.
+        If 'subsample' is smaller than 1 (for fractions), or smaller than the pixel count, a random subsample
+        of valid points is returned.
 
-        If the raster is not loaded, sampling will be done from disk without loading the entire Raster.
+        If the raster is not loaded, sampling will be done from disk using rasterio.sample after loading only the masks
+        of the dataset.
 
         Formats:
             * `as_array` == False: A vector with dataframe columns ["b1", "b2", ..., "geometry"],
             * `as_array` == True: A numpy ndarray of shape (N, 2 + count) with the columns [x, y, b1, b2..].
 
+        :param data_column_name: Name of point cloud data column to use, defaults to "b1".
+        :param data_band: (Only for multi-band rasters) Band to use for data column, defaults to first. Band counting
+            starts at 1.
+        :param store_auxiliary_bands: (Only for multi-band rasters) Whether to save other bands as auxiliary data
+            columns, defaults to False.
         :param subsample: Subsample size. If > 1, parsed as a count, otherwise a fraction.
         :param as_array: Return an array instead of a vector.
-        :param pixel_offset: The point at which to associate the pixel coordinate with ('corner' == upper left).
+        :param random_state: Random state or seed number.
+        :param force_pixel_offset: Force offset to derive point coordinate with. Raster coordinates normally only
+            associate to upper-left corner "ul" ("Area" definition) or center ("Point" definition).
 
         :raises ValueError: If the sample count or fraction is poorly formatted.
 
-        :returns: A GeoDataFrame, or ndarray of the shape (N, 2 + count) where N is the sample count.
+        :returns: A point cloud, or ndarray of the shape (N, 2 + count) where N is the sample count.
         """
-        data_size = self.width * self.height
 
-        # Validate the sample argument.
-        if subsample <= 0.0:
-            raise ValueError(f"subsample cannot be zero or negative (given value: {subsample})")
-        # If the sample is equal to or less than 1, it is assumed to be a fraction.
-        if subsample <= 1.0:
-            sample = int(data_size * subsample)
-        else:
-            sample = int(subsample)
-        if sample > data_size:
-            raise ValueError(f"sample cannot exceed the size of the dataset ({sample} vs {data_size})")
+        # Input checks
+        if not isinstance(data_column_name, str):
+            raise ValueError("Data column name must be a string.")
+        if not isinstance(data_band, int) and (1 < data_band or self.count < data_band):
+            raise ValueError(f"Data band number must be between 1 and the total number of bands of {self.count}.")
```

Comment:
Rephrase "of {self.count}" into "({self.count})".

Reply:
Done